doc/sphinx/proofs/writing-proofs/equality.rst

```diff
@@ -652,6 +650,10 @@ which reduction engine to use.  See :ref:`type-cast`.)  For example:
    If :g:`c` is transparent then it replaces :g:`c` with its
    definition and reduces again until no further reduction is possible.
 
+   In the term :n:`t__1 ... t__n`, :n:`t__1` is the :gdef:`head` of the term.
+   In a term with the form :n:`c t__1 ... t__n`, where :n:`c` is a :term:`constant`,
+   :n:`c` is the :gdef:`head constant`.
```

Comment:
I'm fine with not defining "head" at all (only defining "head constant"). But if we define "head", we should specify that "t__1" is not an application.

Reply:
I'm not sure I like this definition of "head".  Shouldn't "head normal form" have a simple definition in terms of "head"?

Reply:
@JasonGross: can you suggest some wording?

Reply:
> My point is just that if you have `(foo bar) 1 2 3`, you do not call `foo bar` the head.

I don't see why "head" is any different from "head of a list", i.e. the first element.  It seems rather quirky to say it's the head only if it's not applicative.  And thinking about it slightly more, aren't all expressions curried, so `foo bar 1 2 3` means exactly `((((foo bar) 1) 2) 3)`, as does `(foo bar) 1 2 3`?  What am I missing here?

But I can certainly drop the definition of "head".

@JasonGross any suggestion?

Reply:
> And thinking about it slightly more, aren't all expressions curried, so `foo bar 1 2 3` means exactly `((((foo bar) 1) 2) 3)`, as does `(foo bar) 1 2 3`?

That's my point. The three expressions are exactly the same. So you drop all parentheses and take the head of the list and this is the head of the term (and the rest are arguments).

Reply:
I dropped the definition of "head".

Reply:
This makes the refman compilation fail (because the head glossary term is referenced).

Reply:
OK, maybe I should keep the definition then.  How can I exclude applications without having to define that term?

Reply:
You can refer to the [`term_application`](https://coq.inria.fr/refman/language/core/assumptions.html#grammar-token-term_application) grammar token.